languages/csharp/exercises/concept/regular-expressions/.docs/instructions.md

```diff
@@ -0,0 +1,100 @@
+This exercise addresses the parsing of log files.
+
+After a recent security review you have been asked to clean up the organisation's archived log files.
+
+All strings passed to the methods of guaranteed to be non-null and leading and trailing spaces are guaranteed to be removed.
+
+### 1. Identify garbled log lines
+
+You need some idea of how many log lines in your archive do not comply with current standards. You believe that a simple test reveals whether a log line is valid. To be considered valid a line should begin with one of the following strings:
+
+- [TRC]
+- [DBG]
+- [INF]
+- [WRN]
+- [ERR]
+- [FTL]
+
+Implement the `LogParser.IsValidLine()` method to return `false` if a string is not valid otherwise `true`.
+
+```csharp
+var lp = new LogParser();
+lp.IsValidLine("[ERR] A good error here");
+// => true
+lp.IsValidLine("Any old [ERR] text");
+// => false
+lp.IsValidLine("[BOB] Any old text");
+// => false
+```
+
+### 2. Split the log line
+
+A new team has joined the organization and you find their log files are using a strange separator for "fields". Instead of something sensible like a colon ":" they use a string such as "<--->" or "<=>" (because it's prettier) in fact any string that has a first character of "<" and a last character of ">" and any combination of the following "^\*=-" in between.
+
+Implement the `LogParser.SplitLogLIne()` method that takes a line and returns an array of strings each of which contains a field.
+
+```csharp
+var lp = new LogParser();
+lp.IsValidLine("Section 1<===>Section 2<^-^>Section 3");
+// => {"Section 1", "Section 2", "Section 3"}
+```
+
+### 3. Count the number of lines containg a password
+
+It is important to find any passwords included in a file. These will be dealt with automatically but the team needs to know how about passwords occurred in quoted text so that they can be examined manually.
+
+Implement the `LogParser.CountQuotedPasswords()`method to provide an indication of the likely scale of the manual exercise.
+
+The "password" string may be in upper or lower case or any combination.
+
+Lines passed to the routine may or may not be valid as defined in task 1. We process them in the same way whether or not they are valid.
+
+```csharp
+string[] lines =
+{
+    string.Empty,
+    "[INF] passWord ",
+    "\"passWord\"",
+    "[INF] \"The secret password was added by the user\""
+};
+var lp = new LogParser();
+lp.CountQuotedPasswords(lines);
+// => 2
+```
```

Comment:
I think the example needs updating, as the lines are no longer an array but a single string.

Reply:
Done